tests/test_properties.py

```diff
@@ -0,0 +1,383 @@
+        assert np.all(
+            np.linalg.norm(mean_posterior - maximum_likelihood)
+            - np.linalg.norm(mean_prior - maximum_likelihood)
+            < epsilon
+        )
+        assert np.all(
+            np.linalg.norm(mean_prior - mean_posterior)
+            - np.linalg.norm(mean_prior - maximum_likelihood)
+            < epsilon
+        )
+
+        # Posterior parameter estimates improve with increased trust in observations
+        assert np.all(
+            np.linalg.norm(mean_posterior - maximum_likelihood)
+            - np.linalg.norm(previous_mean_posterior - maximum_likelihood)
+            < epsilon
+        )
+
+        previous_mean_posterior = mean_posterior
+
+    # At strong beliefs, we should be close to the maximum likelihood estimate
+    assert np.all(
+        np.linalg.norm(previous_mean_posterior - maximum_likelihood) < epsilon
+    )
+
+
+def test_that_multiple_iterations_of_ies_reduces_generalized_variance():
+    def poly(obs_times):
+        poly = np.stack([np.power(obs_times, 2), obs_times, np.ones(num_obs)])
+        assert poly.shape == (num_params, num_obs)
+        return poly
+
+    ensemble_size = 100
+    num_params = 3
+    num_obs = 4
+
+    prior_es = rng.standard_normal(size=(num_params, ensemble_size))
+
+    obs_times = np.arange(num_obs)
+
+    response_ensemble = poly(obs_times).T @ prior_es
+    assert response_ensemble.shape == (num_obs, ensemble_size)
+
+    # Pick one realization as the truth and add noise to it to create synthetic observations.
+    true_idx = 5
+    obs_errors = rng.normal(scale=2, size=num_obs)
+    obs_values = response_ensemble[:, true_idx] + obs_errors
+
+    noise = rng.standard_normal(size=(num_obs, ensemble_size))
+
+    # Remove truth from ensemble
+    response_ensemble = np.delete(response_ensemble, true_idx, axis=1)
+    noise = np.delete(noise, true_idx, axis=1)
+    prior_es = np.delete(prior_es, true_idx, axis=1)
+    # Update ensemble size because we removed a realization
+    ensemble_size = response_ensemble.shape[1]
+
+    smoother_es = ies.EnsembleSmoother()
+    smoother_es.fit(
+        response_ensemble,
+        obs_errors,
+        obs_values,
+        noise=noise,
+    )
+    posterior_es = smoother_es.update(prior_es)
+    assert np.linalg.det(np.cov(prior_es)) > np.linalg.det(np.cov(posterior_es))
+
+    params_ies = prior_es.copy()
+    responses_ies = response_ensemble.copy()
+
+    smoother_ies = ies.IterativeEnsembleSmoother(ensemble_size)
+    for _ in range(10):
+        smoother_ies.fit(
+            responses_ies,
+            obs_errors,
+            obs_values,
+            noise=noise,
+        )
+        params_ies = smoother_ies.update(params_ies)
+        assert np.linalg.det(np.cov(prior_es)) > np.linalg.det(np.cov(params_ies))
+        # Run forward model with updated parameters
+        responses_ies = poly(obs_times).T @ params_ies
+    # assert np.isclose(
```

Comment:
Posterior calculated using IES does not seem to tend towards posterior calculated using ES.

Reply:
This was due to the transition_matrix being stored in IterativeEnsembleSmoother and not coefficient_matrix.